core/templates/pages/topic-editor-page/services/topic-editor-routing.service.ts

```diff
@@ -15,102 +15,117 @@
 /**
  * @fileoverview Service that handles routing for the topic editor page.
  */
-require('domain/utilities/url-interpolation.service.ts');
-require('services/page-title.service.ts');
-
-
-angular.module('oppia').factory('TopicEditorRoutingService', [
-  '$location', '$rootScope', '$window', 'PageTitleService',
-  'UrlInterpolationService',
-  function(
-      $location, $rootScope, $window, PageTitleService,
-      UrlInterpolationService) {
-    var MAIN_TAB = 'main';
-    var SUBTOPIC_EDITOR_TAB = 'subtopic_editor';
-    var SUBTOPIC_PREVIEW_TAB = 'subtopic_preview';
-    var TOPIC_PREVIEW_TAB = 'topic_preview';
-    var QUESTIONS_TAB = 'questions';
-    var lastTabVisited = 'main';
-    var lastSubtopicId = null;
-
-    var activeTabName = MAIN_TAB;
-
-    // When the URL path changes, reroute to the appropriate tab in the
-    // topic editor page.
-    $rootScope.$watch(function() {
-      return $location.path();
-    }, function(newPath, oldPath) {
+
+import { Location } from '@angular/common';
+import { Injectable } from '@angular/core';
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { UrlInterpolationService } from 'domain/utilities/url-interpolation.service';
+import { PageTitleService } from 'services/page-title.service';
+import { WindowRef } from 'services/contextual/window-ref.service';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class TopicEditorRoutingService {
+  currentPath: string = '';
+  MAIN_TAB: string = 'main';
+  SUBTOPIC_EDITOR_TAB: string = 'subtopic_editor';
+  SUBTOPIC_PREVIEW_TAB: string = 'subtopic_preview';
+  TOPIC_PREVIEW_TAB: string = 'topic_preview';
+  QUESTIONS_TAB: string = 'questions';
+  lastTabVisited: string = 'main';
+  lastSubtopicId: number = null;
+
+  activeTabName: string = this.MAIN_TAB;
+
+  constructor(
+    private location: Location,
+    private pageTitleService: PageTitleService,
+    private urlInterpolationService: UrlInterpolationService,
+    private windowRef: WindowRef) {
+    this.currentPath = this.location.path();
+    this.location.onUrlChange((newPath: string, state: unknown) => {
```

Comment:
Also, there should be a `newPath = newPath.split('#')[1];` where did that go?

Reply:
Added